components/Document/lib/types.ts

```diff
@@ -1,23 +1,182 @@
-import { Hypothesis } from "~/config/types/hypothesis";
-import { Paper } from "~/config/types/paper";
-import { Post } from "~/config/types/post";
-import { TopLevelDocument } from "~/config/types/root_types";
+import { Hub, parseHub } from "~/config/types/hub";
+import { Purchase } from "~/config/types/purchase";
+import {
+  AuthorProfile,
+  ID,
+  RHUser,
+  UnifiedDocument,
+  parseAuthorProfile,
+  parseUnifiedDocument,
+  parseUser,
+} from "~/config/types/root_types";
+import { parseVote, Vote } from "~/config/types/vote";
+import { formatDateStandard } from "~/config/utils/dates";
+import { emptyFncWithMsg } from "~/config/utils/nullchecks";
 
-interface Props {
+export type PaperFormat = {
+  type: "pdf" | "latex";
+  url: string;
+};
+
+export type DocumentType = "hypothesis" | "paper" | "post" | "question";
+
+export type ReviewSummary = {
+  count: number;
+  averageRating: number;
+};
+
+export type ApiDocumentType = "researchhub_post" | "paper" | "hypothesis";
+
+export interface GenericDocument {
```

Comment:
- This is replace ES6 Paper, Hypothesis, Post types.
- Will remove ES6 types when this project is completed